### ISSUE TITLE
Flatten out underlyingErrors

### ISSUE THREAD

**Issue by bobgodwinx** [RxSwiftCommunity/Action#119]
### Currently 

The current implementation of `Action` doesn't allow you to receive the `Swift.Error` when `.execution` is called. But we get an `ActionError` type which is not really useful because you always have to switch within it to the `underlyingError`.

### Proposal 

I think it would be nice for the framework to have something like `var underlyingErrors: Observable<Swift.Error> ` which will only emit when `.execution` actually has an error

**Comment by ashfurrow** [comment]
That makes sense to me, I believe ReactiveSwift's `Command` type has something similar. Let's do it!

**Comment by dangthaison91** [comment]
Should `.errors` will be `Observable<Swift.Error>` while `.executionErrors`/`actionErrors` will be `Observable<ActionError>`? @bobgodwinx

**Comment by bobgodwinx** [comment]
@dangthaison91 exactly, the idea is to separate the errors related to the `output` from the `executionErrors`  this will make sure that on `.errors` we will be getting only `Swift.Error`. Do plan on submitting a PR?
